### PR TITLE
feat: Make vite-node parsing behavior consistent with node/tsx/ts-node

### PR DESCRIPTION
**Pull request by rxliuli**
Closes https://github.com/vitest-dev/vitest/issues/3571

### CODE REVIEW

File: packages/vite-node/src/cli.ts [hunk 1]
```diff
@@ -35,10 +45,13 @@ export interface CliOptions {
   mode?: string
   watch?: boolean
   options?: ViteNodeServerOptionsCLI
+  version?: boolean
+  help?: boolean
   '--'?: string[]
 }
 
 async function run(files: string[], options: CliOptions = {}) {
+  // console.log('run', files, options)
```
Comment (sheremet-va) [inline packages/vite-node/src/cli.ts:54]: Looks like you left a comment here

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: packages/vite-node/src/cli.ts (hunk 1 above)
Comment (rxliuli) [reply packages/vite-node/src/cli.ts:54]: Sorry, I forgot to remove the debug code, I'll fix it later

**Comment by rxliuli** [comment]
@sheremet-va What prevents the merge?

**Comment by sheremet-va** [comment]
> @sheremet-va What prevents the merge?

No functional review done yet

**Comment by marviobezerra** [comment]
I’m also waiting for it. Any news on merging/releasing?

File: packages/vite-node/src/cli.ts [hunk 2]
```diff
@@ -48,6 +60,15 @@ async function run(files: string[], options: CliOptions = {}) {
     process.argv = [...process.argv.slice(0, 2), ...(options['--'] || [])]
   }
 
+  if (options.version) {
+    cli.version(version)
+    cli.outputVersion()
+    process.exit(0)
+  }
+  if (options.help) {
+    cli.outputHelp()
```
Comment (sheremet-va) [inline packages/vite-node/src/cli.ts:69]: `--help` doesn't print version anymore

**Review by sheremet-va — CHANGES_REQUESTED** [review]

**Comment by sheremet-va** [comment]
Looks good to me. Only one small problem was found. We can merge it after it is fixed.

**Comment by rxliuli** [comment]
An error occurred, but it doesn't seem to be because of my modification. . .

**Comment by sheremet-va** [comment]
@rxliuli looks like this broke the usage. `vite-node t.js --watch` doesn't work anymore - `--watch` is inside `--`:

```ts
console.log(options)
// { '--': [ '--watch' ] }
```

**Comment by sheremet-va** [comment]
Tests are testing `esno ./path/` instead of `vite-node --h` which would be preferable here.

**Comment by rxliuli** [comment]
> @rxliuli looks like this broke the usage. `vite-node t.js --watch` doesn't work anymore - `--watch` is inside `--`:
> 
> ```ts
> console.log(options)
> // { '--': [ '--watch' ] }
> ```

Yes, this is consistent with the behavior of node/ts-node/tsx

image

> Tests are testing esno ./dist/cli.js instead of vite-node --h which would be preferable here.

I'm not sure how to run the vite-node command in unit tests, maybe I need to use zx to run `pnpm vite-node`?

**Comment by sheremet-va** [comment]
> Yes, this is consistent with the behavior of node/ts-node/tsx

This breaks watch mode - we don't need consistency when our own features are broken. Doesn't `--script` satisfy your needs?

> I'm not sure how to run the vite-node command in unit tests, maybe I need to use zx to run `pnpm vite-node`?

I will push new utilities in a few minutes.

**Comment by sheremet-va** [comment]
I pushed utilities here: https://github.com/vitest-dev/vitest/pull/3842

Please, open a new PR that specifies the changes in the README. "consistent with node" is not a change - please, reference all breaking changes that you introduced and how to migrate.

**Pull request merged**